src/compiler/compile/nodes/Element.ts

```diff
@@ -555,6 +555,69 @@ export default class Element extends Node {
 				});
 			}
 		}
+
+		if (handlers_map.has('click')) {
+
+			const a11y_interactive = new Set([
+				'a',
+				'button',
+				'select'
+			]);
+
+			if (a11y_interactive.has(this.name)) {
+				return;
+			}
+
+			if (this.name === 'input') {
+				const input_type = attribute_map.get('type');
+				const input_type_interactive = new Set(['button', 'checkbox', 'color', 'file', 'image', 'radio', 'reset', 'submit']);
+
+				if (input_type && input_type_interactive.has(input_type.get_static_value())) {
```

Comment:
I think we should also bail if the value can't be determined statically to not show false positives. Same for line 586/594/601

Reply:
Sorry, I am unsure what you mean exactly? Which value?

Reply:
If someone does `<input type={someVariable} />` and the possible values of `someVariable` are `checkbox` and `radio` but the Svelte compiler can't determine that, it would be a false positive to issue this warning. Therefore I suggest to err on the site of false negatives and don't issue a warning here - similar to how it's done with spread attributes.

Reply:
Thanks for the explanation!
I pushed another commit and added test cases for dynamic `type`, `aria-hidden` and `role` values.